### PR TITLE
Assert schema on TableInfo instead of IndexMetadata mapping in ColumnPolicyIntegrationTest

### PR DESCRIPTION
**Pull request by mfussenegger**
Relates to https://github.com/crate/crate/issues/11939

### CODE REVIEW

**Review by matriv — APPROVED** [review]

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java [hunk 1]
```diff
@@ -323,26 +309,33 @@ public void testCreateTableWithIndex() throws Exception {
...
+        );
+        DocTableInfo table = getTable("novels");
+        assertThat(table.indexColumns()).satisfiesExactly(
+            x -> assertThat(x)
+                .hasName("title_desc_fulltext")
+                .hasType(DataTypes.STRING)
+                .hasAnalyzer("stop")
+                .hasSourceColumnsSatisfying(
+                    s -> assertThat(s).hasName("title"),
+                    s -> assertThat(s).hasName("description")
```
Comment (matriv) [inline server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:328]: Don't know if it's important but we don't assert now:
- oid
- position
- sources

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java [hunk 2]
```diff
@@ -869,29 +857,21 @@ public void testCreateTableWithGeneratedColumn() throws Exception {
...
-                "generated_columns", Map.of(
-                    "day", "date_trunc('day', ts)"
-                )
-            ),
-            "properties", Map.of(
-                "ts", Map.of(
-                    "type", "date",
-                    "position", 1,
-                    "oid", 1,
-                    "format", "epoch_millis||strict_date_optional_time"
```
Comment (matriv) [inline server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:884]: I guess this is always the same for timestamp fields, no need to assert, right?

**Review by matriv — APPROVED** [review]
Thx! Left a minor comments/questions.

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:884]: yes, we don't expose the information in the same way

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:328]: The source is just a link to the actual column so it shouldn't be important here.

File: server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java [hunk 3]
```diff
@@ -497,15 +498,15 @@ public void test_add_column_fixes_inner_types_of_all_its_parents() throws Except
             .setInnerType("a1", DataTypes.INTEGER).build();
 
         assertThat(newTable.getReference(ColumnIdent.of("o", List.of("o", "o"))))
-            .isReference()
+
```
[the remark this answers is not in the document]
Comment (mfussenegger) [reply server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java]: Thanks. I had just done a `:cdo substitute /.isReference()//` and missed those spots.

**Pull request merged**